services/tables/src/main/java/com/linkedin/openhouse/tables/controller/TablesController.java

```diff
@@ -291,4 +292,39 @@ public ResponseEntity<GetAclPoliciesResponseBody> getAclPoliciesForUserPrincipal
     return new ResponseEntity<>(
         apiResponse.getResponseBody(), apiResponse.getHttpHeaders(), apiResponse.getHttpStatus());
   }
+
+  @Operation(
+      summary = "Update locked state for Table",
+      description = "Updates locked state for a table identified by databaseId and tableId",
+      tags = {"Table"})
+  @ApiResponses(
+      value = {
+        @ApiResponse(responseCode = "204", description = "lock PATCH: NO_CONTENT"),
+        @ApiResponse(responseCode = "400", description = "lock PATCH: BAD_REQUEST"),
+        @ApiResponse(responseCode = "401", description = "lock PATCH: UNAUTHORIZED"),
+        @ApiResponse(responseCode = "403", description = "lock PATCH: FORBIDDEN"),
+        @ApiResponse(responseCode = "404", description = "lock PATCH: TABLE_NOT_FOUND")
+      })
+  @PatchMapping(
+      value = {
+        "/v0/databases/{databaseId}/tables/{tableId}/lock",
+        "/v1/databases/{databaseId}/tables/{tableId}/lock"
+      },
+      produces = {"application/json"})
+  public ResponseEntity<Void> updateLock(
```

Comment:
```suggestion
  public ResponseEntity<Void> toggleLock(
```
Toggle or flip lock both works.